spec/metanorma_spec.rb

```diff
@@ -203,6 +138,64 @@ def ruby(x)
 
   it "gives version information" do
     stdout = `metanorma -v -t iso`
-    expect(stdout).to match /Metanorma::ISO \d/
+    expect(stdout).to match(/Metanorma::ISO \d/)
+  end
+
+  it "exports assets" do
+    create_clean_test_files ASCIIDOC_CONFIGURED_HDR
+
+    compile_doc("test.adoc", "-x xml -t iso -e extract,sourcecode")
+    output_file = "extract/sourcecode/sourcecode-0000.txt"
+
+    expect_files_to_not_exists("extract/image/image-0000.png")
+    expect(file_content(output_file, false) + "\n").to eq(code_block)
+  end
+
+  def code_block
+    <<~OUTPUT
+      def ruby(x)
+        if x < 0 && x > 1
+          return
+        end
+      end
+    OUTPUT
+  end
+
+  def file_content(file, temp = true)
+    File.read(file_path(file, temp))
+  end
+
+  def cleanup_test_files
+    FileUtils.rm_f(Dir.glob(file_path("test**")))
+  end
+
+  def file_path(file, temp = true)
+    temp == true ? output_path.join(file) : file
+  end
+
+  def source_doc_file
+    @source_doc_file ||= output_path.join("test.adoc")
+  end
+
+  def create_clean_test_files(content)
+    cleanup_test_files
+    File.write(source_doc_file, content, encoding: "UTF-8")
+  end
+
+  def output_path
+    @output_path ||= Metanorma::Cli.root_path.join("tmp", "acceptance")
```

Comment:
Any reason we are not using Ruby tmpdir?
https://ruby-doc.org/stdlib-2.5.1/libdoc/tmpdir/rdoc/Dir.html

Reply:
Good catch, I was testing it out with with local tmp directory for predictable output but I will change it to use the tmp directory next.